testing/src/main/groovy/org/cristalise/kernel/test/scenario/AgentPredefinedStepsSpecs.groovy

```diff
@@ -87,19 +84,52 @@ class AgentPredefinedStepsSpecs extends Specification implements CristalTestSetu
         newAgent.getRoles().length == 0
     }
 
-    def 'SetAgentPassword can be called by admin to change password for any agent'() {
-        when:
+    def 'SetAgentPassword can be called by admin or by the agent itself'() {
+        when: 'Admin changes the password of triggerAgent'
         String[] params = [ 'test', timeStamp ]
         def triggerAgent = Gateway.getProxyManager().getAgentProxy( Gateway.getLookup().getAgentPath('triggerAgent') )
         assert ! triggerAgent.getPath().isPasswordTemporary()
         agent.execute(triggerAgent, SetAgentPassword.class, params)
 
-        then:
+        then: 'The password becomes temporary'
         Gateway.getLookup().getAgentPath('triggerAgent').isPasswordTemporary()
-        Gateway.getSecurityManager().authenticate('triggerAgent', timeStamp, null).getPath().isPasswordTemporary()
+        Gateway.getSecurityManager().authenticate('triggerAgent', timeStamp, null)
+
+        when: 'triggerAgent changes its own the password'
+        params = [ timeStamp, 'test' ]
+        triggerAgent.execute(triggerAgent, SetAgentPassword.class, params)
+
+        then: 'The password is NOT temporary anymore'
+        ! Gateway.getLookup().getAgentPath('triggerAgent').isPasswordTemporary()
+        Gateway.getSecurityManager().authenticate('triggerAgent', 'test', null)
     }
 
-//    def 'SetAgentPassword can be called by agent to change its own password'() {
-//    }
+    def 'Sign can be used to create a SimpleElectronicSignature record'() {
+        given:
+        String role = "TestRole-$timeStamp()"
+        String name = "TestAgent-$timeStamp()"
+
+        createRole(role)
+        AgentProxy newAgent = createAgent(name, role)
 
+        when:
+        StringBuffer xml = new StringBuffer("<SimpleElectonicSignature>");
+        xml.append("<AgentName>").append(name)      .append("</AgentName>");
+        xml.append("<Password>") .append("dummepwd").append("</Password>");
```

Comment:
typo: should be `dummy`

Reply:
that is the password created in the code :)